### ISSUE TITLE
useless_attribute has a false positive when re-exporting deprecated items.

### ISSUE THREAD

**Issue by Nemo157** [Manishearth/rust-clippy#1372]
Given a crate with the following `src/lib.rs`:

```rust
mod foo { #[deprecated] pub struct Bar; }
pub use foo::Bar;
```

compiling with `cargo build` gives a warning (and `cargo clippy` gives no additional warnings):

```
warning: use of deprecated item, #[warn(deprecated)] on by default
 --> src/lib.rs:2:9
  |
2 | pub use foo::Bar;
  |         ^^^^^^^^
```

To fix this builtin warning you just need to add an `#[allow(deprecated)]` on the `pub use`:

```rust
mod foo { #[deprecated] pub struct Bar; }
#[allow(deprecated)]
pub use foo::Bar;
```

Now `cargo build` gives no warnings, but `cargo clippy` gives a `useless_attribute` warning on the `#[allow(deprecated)]`:

```
warning: useless lint attribute, #[warn(useless_attribute)] on by default
 --> src/lib.rs:2:1
  |
2 | #[allow(deprecated)]
  | ^^^^^^^^^^^^^^^^^^^^
  |
help: if you just forgot a `!`, use
  | #![allow(deprecated)]
  = help: for further information visit https://github.com/Manishearth/rust-clippy/wiki#useless_attribute
```

I just tried updating my nightly toolchain and installing clippy 0.0.103, but got compilation issues, so this was tested with versions:
 * ~cargo 0.13.0-nightly (109cb7c 2016-08-19)~
 * ~rustc 1.12.1 (d4f39402a 2016-10-19)~ (forgot to override the toolchain when recording these versions and no longer have my old nightly toolchain...)
 * clippy 0.0.97

Update:
Got a new nightly toolchain setup and checked that I got the same results with:
 * cargo 0.16.0-nightly (d1bfc01 2016-11-18)
 * rustc 1.15.0-nightly (3bf2be9ce 2016-11-22)
 * clippy 0.0.102

**Comment by Nemo157** [comment]
I just had a look at the code and this seems super-simple to fix so I'll try and throw up a PR in a couple.
